src/algebra/order/monoid_lemmas_zero_lt.lean

```diff
@@ -217,4 +217,369 @@ begin
+  {a b c : α} (ha : a < 1) (bc : b < c) (b0 : 0 < b) : a * b < c :=
+calc  a * b < 1 * b : mul_lt_mul_right' ha b0
+        ... = b     : one_mul b
+        ... < c     : bc
+
+/-- Assumes right covariance. -/
+lemma right.mul_le_one_of_le_of_le [mul_pos_mono α]
+  {a b : α} (ha : a ≤ 1) (hb : b ≤ 1) (b0 : 0 < b) : a * b ≤ 1 :=
+mul_le_of_le_one_of_le ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.mul_lt_one_of_lt_of_le [mul_pos_strict_mono α]
+  {a b : α} (ha : a < 1) (hb : b ≤ 1) (b0 : 0 < b) : a * b < 1 :=
+mul_lt_of_lt_one_of_le ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.mul_lt_one_of_le_of_lt [mul_pos_mono α]
+  {a b : α} (ha : a ≤ 1) (hb : b < 1) (b0 : 0 < b) : a * b < 1 :=
+mul_lt_of_le_one_of_lt ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.mul_lt_one_of_lt_of_lt [mul_pos_strict_mono α]
+  {a b : α} (ha : a < 1) (hb : b < 1) (b0 : 0 < b) : a * b < 1 :=
+mul_lt_of_lt_one_of_lt ha hb b0
+
+-- Lemmas in the form of `1 ≤ a → b ≤ c → 0 < c → b ≤ a * c`,
+-- which assume right covariance.
+
+-- proven with `c0 : 0 ≤ c` as `le_mul_of_one_le_of_le'`
+lemma le_mul_of_one_le_of_le [mul_pos_mono α]
+  {a b c : α} (ha : 1 ≤ a) (bc : b ≤ c) (c0 : 0 < c) : b ≤ a * c :=
+calc  b ≤ c     : bc
+    ... = 1 * c : (one_mul c).symm
+    ... ≤ a * c : mul_le_mul_right' ha c0
+
+lemma lt_mul_of_one_lt_of_le [mul_pos_strict_mono α]
+  {a b c : α} (ha : 1 < a) (bc : b ≤ c) (c0 : 0 < c) : b < a * c :=
+calc  b ≤ c     : bc
+    ... = 1 * c : (one_mul c).symm
+    ... < a * c : mul_lt_mul_right' ha c0
+
+lemma lt_mul_of_one_le_of_lt [mul_pos_mono α]
+  {a b c : α} (ha : 1 ≤ a) (bc : b < c) (c0 : 0 < c) : b < a * c :=
+calc  b < c     : bc
+    ... = 1 * c : (one_mul c).symm
+    ... ≤ a * c : mul_le_mul_right' ha c0
+
+lemma lt_mul_of_one_lt_of_lt [mul_pos_strict_mono α]
+  {a b c : α} (ha : 1 < a) (bc : b < c) (c0 : 0 < c) : b < a * c :=
+calc  b < c     : bc
+    ... = 1 * c : (one_mul c).symm
+    ... < a * c : mul_lt_mul_right' ha c0
+
+/-- Assumes right covariance. -/
+lemma right.one_le_mul_of_le_of_le [mul_pos_mono α]
+  {a b : α} (ha : 1 ≤ a) (hb : 1 ≤ b) (b0 : 0 < b) : 1 ≤ a * b :=
+le_mul_of_one_le_of_le ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.one_lt_mul_of_lt_of_le [mul_pos_strict_mono α]
+  {a b : α} (ha : 1 < a) (hb : 1 ≤ b) (b0 : 0 < b) : 1 < a * b :=
+lt_mul_of_one_lt_of_le ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.one_lt_mul_of_le_of_lt [mul_pos_mono α]
+  {a b : α} (ha : 1 ≤ a) (hb : 1 < b) (b0 : 0 < b) : 1 < a * b :=
+lt_mul_of_one_le_of_lt ha hb b0
+
+/-- Assumes right covariance. -/
+lemma right.one_lt_mul_of_lt_of_lt [mul_pos_strict_mono α]
+  {a b : α} (ha : 1 < a) (hb : 1 < b) (b0 : 0 < b) : 1 < a * b :=
+lt_mul_of_one_lt_of_lt ha hb b0
+
+-- proven with `a0 : 0 ≤ a` as `mul_le_of_le_one_right'`
+lemma mul_le_of_le_one_right [pos_mul_mono α] {a b : α} (h : b ≤ 1) (a0 : 0 < a) :
+  a * b ≤ a :=
+mul_le_of_le_of_le_one le_rfl h a0
```

Comment:
I suggest opening a separate PR for this issue, in case you actually want to change the name of this lemma.  Already as is, this PR is very long and will take time to review.